src/main/java/Duke.java

```diff
@@ -1,10 +1,52 @@
+import java.util.Scanner;
+
 public class Duke {
+    protected static FileManager storage;
+
+    public Duke() {
+        // set up the things needed to start Duke
+        FileManager.createFolder();
+        storage = new FileManager();
+        if (storage.retrieveTextFile()) {
+            storage.loadData();
+        }
+    }
+
+    public void run() {
```

Comment:
Good use of SLAP in this function! I like that you extracted the handlers into methods. :)